mne/viz/_3d.py

```diff
@@ -966,6 +966,99 @@ def _limits_to_control_points(clim, stc_data, colormap):
     return ctrl_pts, colormap
 
 
+def _handle_time(time_label, time_unit, times):
+    """Handle time label string and units."""
+    if time_label == 'auto':
+        time_label = 'time=%0.3f ms'
+    if time_unit == 's':
+        times = times
+    elif time_unit == 'ms':
+        times = 1e3 * times
+    else:
+        raise ValueError("time_unit needs to be 's' or 'ms', got %r" %
+                         (time_unit,))
+
+    return time_label, times
+
+
+def _plot_mpl_stc(stc, subject=None, surface='inflated', hemi='lh',
+                  colormap='auto', time_label='auto', smoothing_steps=10,
+                  alpha=1.0, subjects_dir=None, views='lat', figure=None,
+                  initial_time=0, time_unit='s'):
+    """Plot source estimate using mpl."""
+    import matplotlib.pyplot as plt
+    from mpl_toolkits.mplot3d import Axes3D, art3d
+    from matplotlib import cm
+    import nibabel as nib
+    from surfer.utils import smoothing_matrix
```

Comment:
the smoothing matrix code is IIRC equivalent to our smoothing matrix code, can you see if you can do it using ours?

Plus I don't think `surfer` imports will work at all unless PySurfer works completely